### ISSUE TITLE
Defining the values but inferring the keys of an object

### ISSUE THREAD

**Issue by STRML** [facebook/flow#3661]
This [doesn't throw any errors](https://flow.org/try/#0C4TwDgpgBAshDO8CGBzaBeKBvAUFKwAlsADYQBcU8wAToQHYoA0eUARgPYAmIA-JdTqMcAXwDcOHAGMO9alABmHDlEy58AciUcNlLEVIUoW5QBViZDSJaa2SGruwGylDXZrnDV0dNnz3egDaANYQIJQAVAC6lHCIqBAiqorKEjgKAK70UkSyUGjAAGpIJBkQAKr0oJBcABSEXJQAJADSYfAAPNUQHAopHAB8AJTYrDQQwBk09P2BDVESIpKZ2bkzBcWlEKbgEHUNzW0gnd297PbDo-jjk9PnNHNcCz44GyVlld11bkgAXhpDV4TTZlHY1Wo-f4jAD00KgAHcOBkSFwoCRCKECAALQjwAgqCA0GgcGhMKBcDgIKD0DjAIA):

```jsx
type Message = {
  title: string,
  body?: string
};

const foo = {
  'foo': {title: 'fooTitle'},
  'bar': {title: 'barTitle'}
}
const bar: {[key: *]: Message} = foo;

function getValueUntyped(id: $Keys<typeof foo>) {
  return foo[id];
}

function getValueTyped(id: $Keys<typeof bar>) {
  return bar[id];
}

getValueUntyped('baz')
getValueTyped('baz') // would like this to error, does not
```

The existential type in the definition of `foo` should allow flow to keep its own key inference, but it is lost somewhere.

**Comment by vkurchatkin** [comment]
* is inferred as `string`, so this seems fine

**Comment by STRML** [comment]
Right, but I want it to be inferred as `'foo.bar'`; is there any way to do that?

**Comment by vkurchatkin** [comment]
I don't think so, for the same reason this doesn't works:

```js
const xx: * = 'foo';
const yy: typeof xx = 'bar';
```

Flow doesn't infer string literals, it infers string with known value, which is subsequently ignored by `typeof`

**Comment by SamChou19815** [comment]
Closing since * support was removed
